src/main/java/com/shapesecurity/shift/reducer/CloneReducer.java

```diff
@@ -131,7 +131,7 @@ public Expression reduceCompoundAssignmentExpression(@NotNull CompoundAssignment
     @NotNull
     @Override
     public MemberExpression reduceComputedMemberExpression(@NotNull ComputedMemberExpression node, @NotNull Node object, @NotNull Node expression) {
-        return new ComputedMemberExpression((Expression) object, (ExpressionSuper) expression);
+        return new ComputedMemberExpression((Expression) expression, (ExpressionSuper) object);
```

Comment:
I agree that also standardizing the interface to the AST node constructors would also be valuable, but there's a lot less potential for confusion there, since those are typed. Also, I don't have the file which generated them.